### PR TITLE
Fix login using new BGG endpoint

### PR DESCRIPTION
**Pull request by tvedeane**

### CODE REVIEW

File: app/src/main/java/com/boardgamegeek/io/RetryInterceptor.kt [hunk 1]
```diff
@@ -56,7 +56,10 @@ class RetryInterceptor : Interceptor {
 
     private fun nextBackOffMillis(response: Response): Long {
         return when (response.code) {
-            COLLECTION_REQUEST_PROCESSING -> backOff202.nextBackOffMillis()
+            COLLECTION_REQUEST_PROCESSING -> {
+                if (retry202Response) backOff202.nextBackOffMillis()
+                else BackOff.STOP
+            }
```
Comment (tvedeane) [inline app/src/main/java/com/boardgamegeek/io/RetryInterceptor.kt:62]: Not very clean change I have to say. We could consider changing this by splitting `RetryInterceptor` into 3 classes (`Retry202Interceptor`, `Retry429Interceptor`, `Retry503Interceptor`).

Comment (tvedeane) [reply app/src/main/java/com/boardgamegeek/io/RetryInterceptor.kt:62]: Yes, exactly - login is returning 202 after sending correct username/password pair.

Comment (ccomeaux) [reply app/src/main/java/com/boardgamegeek/io/RetryInterceptor.kt:62]: Oh, I see. I think there is a better way of doing this, but in the name of fixing it, I'll likely just go ahead and merge what you have here. There's nothing wrong with it.

Comment (tvedeane) [reply app/src/main/java/com/boardgamegeek/io/RetryInterceptor.kt:62]: Cool, thanks Chris.

**Review by ccomeaux — APPROVED** [review]

**Pull request merged**